lib/membrane_rtmp_plugin/rtmp/source/message_validator.ex

```diff
@@ -14,30 +14,30 @@ defmodule Membrane.RTMP.MessageValidator do
       alias Membrane.RTMP.Messages
 
       @impl true
-      def validate_release_stream(%Messages.ReleaseStream{}), do: :ok
+      def validate_release_stream(%Messages.ReleaseStream{}), do: {:ok, "release stream success"}
 
       @impl true
-      def validate_publish(%Messages.Publish{}), do: :ok
+      def validate_publish(%Messages.Publish{}), do: {:ok, "publish success"}
 
       @impl true
-      def validate_set_data_frame(%Messages.SetDataFrame{}), do: :ok
-
+      def validate_set_data_frame(%Messages.SetDataFrame{}), do: {:ok, "set data frame success"}
       defoverridable Membrane.RTMP.MessageValidator
     end
   end
 
   @doc """
   Validates the `t:Membrane.RTMP.Messages.ReleaseStream.t/0` message.
   """
-  @callback validate_release_stream(Messages.ReleaseStream.t()) :: :ok | {:error, any()}
+  @callback validate_release_stream(Messages.ReleaseStream.t()) ::
+              {:ok | :error, message :: any()}
 
   @doc """
   Validates the `t:Membrane.RTMP.Messages.Publish.t/0` message.
   """
-  @callback validate_publish(Messages.Publish.t()) :: :ok | {:error, any()}
+  @callback validate_publish(Messages.Publish.t()) :: {:ok | :error, message :: any()}
```

Comment:
II would keep `{:ok, result :: any()} | {:error, reason :: any()}`, it is a little bit more readable than trying to pack it into a single tuple with `|`.

Reply:
And you can probably pack that into `validation_result_t`.